### PR TITLE
Reattempt part upload upon failure

### LINKED ISSUE #2: Reattempt failed upload part instead of crashing immediately
Looks like my upload failed due to a RequestTimeout. If I start the upload again, it does properly check what parts have already been uploaded but it takes a lot of time.

I made some changes locally to make this work, wanted to check if there was interest before sending a PR.

### PR DESCRIPTION
**Pull request by azeemba**
When a part upload fails, we should retry since a lot of the issues
can be transient.

Fixes #2

### CODE REVIEW

**Comment by azeemba** [comment]
Any thoughts on #3 and #4 @tbumi?

File: main.py [hunk 1]
```diff
@@ -193,14 +193,25 @@ def upload_part(byte_pos, vault_name, upload_id, part_size, fileobj, file_size,
     print('Uploading part {0} of {1}... ({2:.2%})'.format(
         part_num + 1, num_parts, percentage))
 
-    response = glacier.upload_multipart_part(
-        vaultName=vault_name, uploadId=upload_id, range=range_header,
-        body=part)
+    MAX_ATTEMPTS = 10
```
Comment (tbumi) [inline main.py]: Constants should be defined outside of functions.

File: main.py [hunk 2]
```diff
@@ -193,14 +193,25 @@ def upload_part(byte_pos, vault_name, upload_id, part_size, fileobj, file_size,
...
+    uploaded = False
+    for i in range(MAX_ATTEMPTS):
```
Comment (tbumi) [inline main.py]: instead of setting variable `uploaded`, how about using `for..else`?

File: main.py [hunk 3]
```diff
@@ -193,14 +193,25 @@ def upload_part(byte_pos, vault_name, upload_id, part_size, fileobj, file_size,
...
+        try:
+            response = glacier.upload_multipart_part(
+                vaultName=vault_name, uploadId=upload_id, range=range_header,
+                body=part)
+            checksum = calculate_tree_hash(part, part_size)
+            assert checksum == response['checksum'], 'Checksums do not match.'
```
Comment (tbumi) [inline main.py]: instead of asserting, just use an `if` condition to check: if checksums match, then break from loop.

**Review by tbumi — CHANGES_REQUESTED** [review]
I left some suggestions, feel free to comment if you have any questions.

File: main.py (hunk 3 above)
Comment (azeemba) [reply main.py]: The previous code was asserting so I stuck with that. Are you sure you want to change that behavior?

Comment (tbumi) [reply main.py]: Yeah, I plan to change some code styles myself anyways.

**Review by tbumi — APPROVED** [review]
Cool. Thanks for the contribution, I will merge this PR.

**Pull request merged**